pkg/systray/runtime/runner.go

```diff
@@ -0,0 +1,234 @@
+package runtime
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// SystrayUserProcessRunner creates a launcher systray process each time it detects
+// a new console (GUI) user. If the current console user's systray process dies, it
+// will create a new one.
+// Initalize with NewSystrayUserProcessRunner().
+type SystrayUserProcessRunner struct {
+	logger            log.Logger
+	executionInterval time.Duration
+	interrupt         chan struct{}
+	// uidProcs is a map of uid to systray process
+	uidProcs map[string]*os.Process
+}
+
+// NewSystrayUserProcessRunner creates and returns a new SystrayUserProcess runner and initializes all required fields
+func NewSystrayUserProcessRunner(logger log.Logger) *SystrayUserProcessRunner {
+	return &SystrayUserProcessRunner{
+		logger:            logger,
+		interrupt:         make(chan struct{}),
+		uidProcs:          make(map[string]*os.Process),
+		executionInterval: time.Second * 5,
+	}
+}
+
+// Execute checks if the current console user has a systray process running. If not, it will start a new one.
+func (r *SystrayUserProcessRunner) Execute() error {
+	f := func() {
+		if err := r.runConsoleUserSystray(); err != nil {
+			level.Error(r.logger).Log(
+				"msg", "error running systray",
+				"err", err,
+			)
+		}
+	}
+
+	f()
+
+	ticker := time.NewTicker(r.executionInterval)
+	for {
+		select {
+		case <-ticker.C:
+			f()
+		case <-r.interrupt:
+			return nil
+		}
+	}
+}
+
+// Interrupt stops creating launcher systray processes and kils the existing ones.
+func (r *SystrayUserProcessRunner) Interrupt(err error) {
+	r.interrupt <- struct{}{}
+	for _, proc := range r.uidProcs {
+		if isExists, _ := processExists(proc.Pid); isExists {
+			if err := proc.Kill(); err != nil {
+				level.Error(r.logger).Log(
+					"msg", "error killing systray",
+					"err", err,
+				)
+			}
+		}
+	}
+}
+
+func (r *SystrayUserProcessRunner) runConsoleUserSystray() error {
+	consoleOwnerUid, err := consoleOwnerUid()
+	if err != nil {
+		return fmt.Errorf("getting console owner uid: %w", err)
+	}
+
+	// there seems to be a brief moment during start up where root has the console
+	// if we spin up the process for root after the user gets the console it will
+	// add another systray icon, so don't spin it up for root
+	if consoleOwnerUid == 0 {
+		level.Info(r.logger).Log(
+			"msg", "skipping systray for root user",
+			"uid", consoleOwnerUid,
+		)
+
+		return nil
+	}
+
+	uid := fmt.Sprintf("%d", consoleOwnerUid)
```

Comment:
well we have to cast something some where

getting the `/dev/console` owner returns a uid
`consoleInfo.Sys().(*syscall.Stat_t).Uid,`

we need a string to look up that user with
`user.LookupId(string)`

I prefer to leave the map as a string since when we add for windows the user ID (SID) will be a string